crates/sui-types/src/object.rs

```diff
@@ -79,14 +80,22 @@ impl MoveObject {
         version: SequenceNumber,
         contents: Vec<u8>,
         protocol_config: &ProtocolConfig,
+        system_mutation: bool,
     ) -> Result<Self, ExecutionError> {
-        Self::new_from_execution_with_limit(
-            type_,
-            has_public_transfer,
-            version,
-            contents,
-            protocol_config.max_move_object_size(),
-        )
+        let bound = if system_mutation {
+            if contents.len() as u64 > protocol_config.max_move_object_size() {
+                warn!(
```

Comment:
can you make this debug_fatal? Then we can observe it with metrics and also catch it in tests

Reply:
done :)